### PR TITLE
80452 Cancel improvements

### PR DESCRIPTION
**Pull request by bratli**
DevOps#80452

-Cancel outlook meeting
-Clear M01 dates if IPO was completed
-Tests
- More validation
- CancelIpoDto

### CODE REVIEW

**Comment by bratli** [comment]
@janid1967 There is a test failing, need to ask you something about it.
Also, would like to discuss how we handle calling to foreign APIs in this command handler

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/TestFactory.cs [hunk 1]
```diff
@@ -418,7 +418,8 @@ private void AddHackerUser(List<ProCoSysProject> commonProCoSysProjects)
                         Permissions.USER_READ,
                         Permissions.IPO_READ,
                         Permissions.IPO_SIGN,
-                        Permissions.IPO_WRITE
+                        Permissions.IPO_WRITE,
+                        Permissions.IPO_CREATE
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/TestFactory.cs:422]: why should completer have create permissions?

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/TestFactory.cs [hunk 2]
```diff
@@ -448,12 +449,13 @@ private void AddHackerUser(List<ProCoSysProject> commonProCoSysProjects)
...
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/TestFactory.cs:453]: why should accepter have create permissions?

**Review by janid1967 — COMMENTED** [review]
I think it's incorrect  to change permissions in TestFactory

File: src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/TestFactory.cs (hunk 2 above)
Comment (bratli) [reply src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/TestFactory.cs:453]: I guess in my head accepter, planner, and completed could be the same person - but I see now that for testing purposes it shouldnt be like that

Comment (janid1967) [reply src/tests/Equinor.ProCoSys.IPO.WebApi.IntegrationTests/TestFactory.cs:453]: Exactly. Then we have to use planner test user to test create IPO endpoint. Other should get access denied

**Review by janid1967 — CHANGES_REQUESTED** [review]
Revert permission changes

File: src/tests/Equinor.ProCoSys.IPO.MainApi.Tests/MainApi/McPkg/MainApiMcPkgServiceTests.cs [hunk 3]
```diff
@@ -120,6 +120,10 @@ public async Task GetMcPkgsByMcPkgNosAsync_ShouldReturnEmptyList_WhenResultIsInv
         public async Task SetM01Dates_ShouldReturnOk_WhenCallSucceeds() 
             => await _dut.SetM01DatesAsync(_plant, 1, "Project1", new List<string>{"A"}, new List<string>());
 
+        [TestMethod]
+        public async Task ClearM01Dates_ShouldReturnOk_WhenCallSucceeds()
+            => await _dut.ClearM01DatesAsync(_plant, 1, "Project1", new List<string> { "A" }, new List<string>());
```
[the remark this answers is not in the document]
Comment (bratli) [reply src/tests/Equinor.ProCoSys.IPO.MainApi.Tests/MainApi/McPkg/MainApiMcPkgServiceTests.cs:125]: True... What/How should this be tested? Other tests in this class tests the response from endpoint. I can test that PutAsync is called for example, and this:

            _foreignApiClient
                .Setup(x => x.PutAsync(It.IsAny<string>(), It.IsAny<HttpContent>()))
                .Returns(Task.FromException(new Exception("Something went wrong")));

            await Assert.ThrowsExceptionAsync<Exception>(() =>
                _dut.SetM01DatesAsync(_plant, 1, "Project1", new List<string> { "A" }, new List<string>()));

[the remark this answers is not in the document]
Comment (janid1967) [reply src/tests/Equinor.ProCoSys.IPO.MainApi.Tests/MainApi/McPkg/MainApiMcPkgServiceTests.cs:125]: Drop tests .. there is almost nothing to test. Testing that put is called has minimum value

**Review by janid1967 — APPROVED** [review]

**Pull request merged**